### ISSUE TITLE
Error Getting Bucket Region

### ISSUE THREAD

**Issue by kouwasi** [deliciousbrains/wp-amazon-s3-and-cloudfront#456]
So got that error 

```
Error creating bucket — Error executing "CreateBucket" on "https://aaabbb.s3.amazonaws.com/"; AWS HTTP error: Client error: `PUT https://aaabbb.s3.amazonaws.com/` resulted in a `400 Bad Request` response: AuthorizationHeaderMalformedThe authorization header (truncated...) AuthorizationHeaderMalformed (client): The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'ap-northeast-1' - AuthorizationHeaderMalformedThe authorization header is malformed; the region 'us-east-1' is wrong; expecting 'ap-northeast-1'ap-northeast-
```

I think so need like this fix
```php

		try {
			$region = defined('AS3CF_REGION') ? AS3CF_REGION : $this->get_s3client()->get_bucket_location( array( 'Bucket' => $bucket ) );
		} catch ( Exception $e ) {
			$error_msg_title = '<strong>' . __( 'Error Getting Bucket Region', 'amazon-s3-and-cloudfront' ) . '</strong> &mdash;';
			$error_msg       = sprintf( __( 'There was an error attempting to get the region of the bucket %s: %s', 'amazon-s3-and-cloudfront' ), $bucket, $e->getMessage() );
			AS3CF_Error::log( $error_msg );
			return new WP_Error( 'exception', $error_msg_title . $error_msg );
		}
```

**Comment by keesvanlierop** [comment]
Running into the same issue here when selecting an existing bucket that is not in the 'us-east-1' region.

**Comment by ianmjones** [comment]
Hey @keesvanlierop, which region is the bucket in?

**Comment by keesvanlierop** [comment]
Mine is in eu-west-1.

For now I have just created a new bucket with a different name through the plugin. That works perfectly fine.

**Comment by ianmjones** [comment]
Hey @kouwasi & @keesvanlierop,

Haven't been able to reproduce this, even though I have oodles of buckets in basically all the regions.

Is there anything "interesting" about your IAM User's policies?

Can you share the JSON version of your IAM User's policy?

Do you have any policies attached to your buckets?

**Comment by kouwasi** [comment]
Hi @ianmjones 
I'm using an AmazonS3FullAccess

**Comment by hayleynewton** [comment]
We are also having this issue in regards to the region.

We created a new instance of wordpress, added IAM with full access and tried to connect to a bucket with no luck. We get the same region error.

We do have one instance setup correctly which has worked when changing buckets but we now get the region error trying to point it to another bucket.

Making a bucket through the plugin works as expected.

**Comment by keesvanlierop** [comment]
> I'm using an AmazonS3FullAccess
Same here.

**Comment by ianmjones** [comment]
Guys & Gals, could really do with the following...

> Can you share the JSON version of your IAM User's policy?

> Do you have any policies attached to your buckets?

Ideally, diagnostic info from Support tab of WP Offload S3 Lite's settings page too. There's obviously something getting in the way that as yet we can't reproduce, more details on the sites that have this problem would likely help.

**Comment by jaredcza** [comment]
Hi,

This issue was occurring for me when attempting to select an existing bucket located in the eu-west-1 region. I spent many hours tearing my hair out! Error below (I have replaced the actual bucket name and URLs with ######).

> Error saving bucket — Error Getting Bucket Region —There was an error attempting to get the region of the bucket ######: Error executing "GetBucketLocation" on "https://######.s3.amazonaws.com/?location"; AWS HTTP error: Client error: `GET https://######.s3.amazonaws.com/?location` resulted in a `400 Bad Request` response: AuthorizationHeaderMalformedThe authorization header (truncated...) AuthorizationHeaderMalformed (client): The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'eu-west-1' - AuthorizationHeaderMalformedThe authorization header is malformed; the region 'us-east-1' is wrong; expecting 'eu-west-1'eu-west-1C5650D3FF59225165I4uqoSsBfHzl14PVG4qAaKFPrFmWkxZFMkx8oQd/gbWHfVzZy06KDe4

I **resolved** this by amending line 98 in amazon-s3-and-cloudfront\classes\providers\aws-provider.php to read:

`protected $default_region = 'eu-west-1';`

**Comment by ianmjones** [comment]
If you're having this issue, please can you check the option_value for your options table record with option_name = "tantan_wordpress_s3" and tell me what the "region" is set to?

Also, does the "bucket" in the same value have the newly selected bucket you wanted or an old one?

**Comment by ianmjones** [comment]
Hey folks, WP Offload S3 Lite 1.4.2 has just been released with a bug fix for this issue.

**Comment by wallter** [comment]
Not solved in v1.4.2: [#460](https://github.com/deliciousbrains/wp-amazon-s3-and-cloudfront/issues/460)
